phpBB/assets/javascript/editor.js

```diff
@@ -226,6 +231,55 @@ function addquote(post_id, username, l_wrote) {
 	return;
 }
 
+/**
+* Create a quote block for given text
+*
+* Possible attributes:
+*   - author:  author's name (usually a username)
+*   - post_id: post_id of the post being quoted
+*   - user_id: user_id of the user being quoted
+*   - time:    timestamp of the original message
+*
+* @param  {!string} text       Quote's text
+* @param  {!Object} attributes Quote's attributes
+* @return {!string}            Quote block to be used in a new post/text
+*/
+function generate_quote(text, attributes)
+{
+	var quote = '[quote';
+	if ('author' in attributes)
+	{
+		// Add the author as the BBCode's default attribute
+		quote += '=' + enquote(attributes.author);
+		delete attributes.author;
+	}
+	for (var name in attributes)
```

Comment:
I added a `hasOwnProperty()` check, does that work for  you?

I fed the updated function to JSHint.com and it didn't output any warnings.